lib/mongoProvider.js

```diff
@@ -103,7 +103,7 @@ MongoCollection.prototype.find = function (query, req) {
     return self.db().then(function (db) {
         return self.beforeFindListeners.fire(query, req).then(function () {
 
-            return q.ninvoke(db.collection(self.name).find(query), "toArray").then(function(res) {
+            return q.ninvoke(db.collection(self.name).find(query, req), "toArray").then(function(res) {
```

Comment:
May I ask what is actually your final intention, what you do? This is quite advanced and I'm just wondering... I'm always curious what users do with jsreport.